### DISCUSSION TITLE
vercel deploy --prebuilt Invalid Json error

### DISCUSSION THREAD

**Discussion by migsdeving** [vercel/community#2157] [Help] [answered]
I'm currently having multiple issues with my build times while generating pages with getStaticProps so I decided to build locally and deploy the prebuilt as mentioned here: https://vercel.com/guides/deploying-locally-built-nextjs

After running `vercel build` successfully, I run `vercel deploy --prebuilt` and it always logs `Error: Invalid JSON`.

When running  `vercel deploy --prebuilt -d` in debug mode, I get the following error:

>Vercel CLI 28.20.0
> [debug] [2023-04-21T23:24:01.984Z] user supplied known subcommand: "deploy"
> [debug] [2023-04-21T23:24:02.088Z] Spinner invoked (Retrieving project…) with a 1000ms delay
> [debug] [2023-04-21T23:24:02.158Z] #1 → GET https://api.vercel.com/v2/user
> [debug] [2023-04-21T23:24:02.163Z] #2 → GET https://api.vercel.com/v8/projects/***********************
> [debug] [2023-04-21T23:24:02.793Z] #1 ← 200 OK: sfo1::******************* [634ms]
> [debug] [2023-04-21T23:24:02.813Z] #2 ← 200 OK: sfo1::************************* [650ms]
> [debug] [2023-04-21T23:24:02.985Z] Spinner invoked (Deploying *******) with a 0ms delay
[client-debug] 2023-04-21T23:24:02.986Z Creating deployment...
[client-debug] 2023-04-21T23:24:02.986Z Provided 'path' is a directory.
[client-debug] 2023-04-21T23:24:02.989Z Found 4 rules in .vercelignore
[client-debug] 2023-04-21T23:24:02.990Z Building file tree...
[client-debug] 2023-04-21T23:24:08.904Z Found 102660 files in the specified directory
[client-debug] 2023-04-21T23:25:12.182Z Yielding a 'hashes-calculated' event with 66222 hashes
[client-debug] 2023-04-21T23:25:12.307Z Using provided API URL: https://api.vercel.com
[client-debug] 2023-04-21T23:25:12.307Z Using provided user agent: vercel 28.20.0 node-v18.12.1 win32 (x64)
[client-debug] 2023-04-21T23:25:12.308Z Setting platform version to harcoded value 2
[client-debug] 2023-04-21T23:25:12.308Z Creating the deployment and starting upload...
[client-debug] 2023-04-21T23:25:12.309Z Determining necessary files for upload...
[client-debug] 2023-04-21T23:25:12.310Z Creating deployment
[client-debug] 2023-04-21T23:25:13.151Z Sending deployment creation API request
[client-debug] 2023-04-21T23:25:47.281Z Deployment response: {"error":{"code":"bad_request","message":"Invalid JSON"}}
[client-debug] 2023-04-21T23:25:47.282Z Error: Deployment request status is 400
[client-debug] 2023-04-21T23:25:47.282Z Yielding a 'error' event
> [debug] [2023-04-21T23:25:47.352Z] Error: Error: Invalid JSON
Error: Invalid JSON
    at Now.handleDeploymentError (C:\Users\migue\AppData\Local\Yarn\Data\global\node_modules\vercel\dist\index.js:209879:25)
    at processDeployment (C:\Users\migue\AppData\Local\Yarn\Data\global\node_modules\vercel\dist\index.js:203036:41)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Now.create (C:\Users\migue\AppData\Local\Yarn\Data\global\node_modules\vercel\dist\index.js:209800:28)
    at createDeploy (C:\Users\migue\AppData\Local\Yarn\Data\global\node_modules\vercel\dist\index.js:202551:16)
    at module.exports.__webpack_modules__.92005.exports.default (C:\Users\migue\AppData\Local\Yarn\Data\global\node_modules\vercel\dist\index.js:193653:22)
    at main (C:\Users\migue\AppData\Local\Yarn\Data\global\node_modules\vercel\dist\index.js:199778:20)
Error: Invalid JSON>

Im on the latest version of the CLI `Vercel CLI 28.20.0`.

I have no idea why this is happening but I've been trying to figure it out for a few days, does somebody know how to fix this ?

Thank you

EDIT: Tested it on a fresh nextjs app and everything worked fine

**Comment by edjonesdev** [comment, marked as answer]
I found that the error had something to do with my serverless functions going over the 50mb limit, and adding this to my `next.config.js` got rid of the error for me:

```js
 experimental: {
    outputFileTracingExcludes: {
      '*': [
        'node_modules/@swc/core-linux-x64-gnu',
        'node_modules/@swc/core-linux-x64-musl',
        'node_modules/@esbuild/linux-x64',
      ],
    },
  },
```
